usr/share/rear/restore/COMMVAULT/default/400_restore_with_galaxy.sh

```diff
@@ -0,0 +1,91 @@
+#
+# restore data with Commvault
+#
+
+# verify that we have a backupset
+[ "$COMMVAULT_BACKUPSET" ]
+StopIfError "Commvault  Backup Set not defined [COMMVAULT_BACKUPSET=]."
+
+# create argument file
+
+cat <<EOF >$TMP_DIR/commvault.restore.options
+$(test -r "$COMMVAULT_Q_ARGUMENTFILE" && cat "$COMMVAULT_Q_ARGUMENTFILE")
+[sourceclient]
+$HOSTNAME
+[level]
+1
+[options]
+QR_PRESERVE_LEVEL
+QR_DO_NOT_OVERWRITE_FILE_ON_DISK
+$COMMVAULT_PIT
+[dataagent]
+Q_LINUX_FS
+[backupset]
+$COMMVAULT_BACKUPSET
+[sourcepaths]
+/
+[destinationpath]
+$TARGET_FS_ROOT
+EOF
+
+if [ "x$COMMVAULT_ZEIT" != "x" ]; then
+cat <<EOF >>$TMP_DIR/commvault.restore.options
+[browseto]
+$COMMVAULT_ZEIT
+EOF
+fi
+
+# initialize variable
+jobstatus=Unknown
+
+if jobid=$(qoperation restore -af $TMP_DIR/commvault.restore.options) ; then
+	jobid=${jobid// /}	# remove trailing blanks
+    	prevstatus=
+
+	LogPrint "Restoring data with Commvault (job $jobid)"
+
+	while true
+	do
+		sleep 60
+		# output of qlist job -co sc -j ## :
+
+        	# STATUS     COMPLETE PERCENTAGE
+        	# ------     -------------------
+        	# Running    5
+
+        	jobstatus=$(/opt/commvault/Base/qlist job -j $jobid -co sc | tail -n 1)
+		StopIfError "Could not receive job details. Check log file."
```

Comment:
Better don't use ...IfError functions in new code.

Reply:
Yes, this is crap because it is unclear what to check for... that the `tail` didn't work? I think better check for `qlist` failing